File: src/main/java/at/fhhagenberg/sqe/di/ElevatorServicesModule.java
package at.fhhagenberg.sqe.di;

import at.fhhagenberg.sqe.IElevator;
import at.fhhagenberg.sqe.api.*;
import com.google.inject.AbstractModule;
import com.google.inject.Singleton;

public class ElevatorServicesModule extends AbstractModule {
    @Override
    protected void configure() {
        bind(BuildingFloorService.class).to(BuildingFloorServiceImpl.class);
        bind(ElevatorControlSystemService.class).to(ElevatorControlSystemServiceImpl.class);
        bind(ElevatorService.class).to(ElevatorServiceImpl.class);
        bind(FloorButtonService.class).to(FloorButtonServiceImpl.class);
        bind(ServicedFloorService.class).to(ServicedFloorServiceImpl.class);
        bind(IElevator.class).to(CachedElevatorControl.class).in(Singleton.class);
    }
}

File: src/main/kotlin/at/fhhagenberg/sqe/di/ElevatorServicesModule.kt
package at.fhhagenberg.sqe.di

import at.fhhagenberg.sqe.api.*
import com.google.inject.AbstractModule
import com.google.inject.Singleton
import sqelevator.CachedElevatorControl
import sqelevator.IElevator

class ElevatorServicesModule : AbstractModule() {
    override fun configure() {
        bind(BuildingFloorService::class.java).to(BuildingFloorServiceImpl::class.java)
        bind(ElevatorControlSystemService::class.java).to(ElevatorControlSystemServiceImpl::class.java)
        bind(ElevatorService::class.java).to(ElevatorServiceImpl::class.java)
        bind(FloorButtonService::class.java).to(FloorButtonServiceImpl::class.java)
        bind(ServicedFloorService::class.java).to(ServicedFloorServiceImpl::class.java)
        bind(IElevator::class.java).to(CachedElevatorControl::class.java).`in`(Singleton::class.java)
    }
}